vividus-engine/src/main/java/org/vividus/replacement/CompositeStepsReplacementStoryReporter.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2019-2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.vividus.replacement;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.jbehave.core.model.Step;
+import org.jbehave.core.steps.StepCreator.StepExecutionType;
+import org.vividus.ChainedStoryReporter;
+import org.vividus.testcontext.TestContext;
+
+public class CompositeStepsReplacementStoryReporter extends ChainedStoryReporter
+{
+    private static final Pattern DEPRECATED_COMPOSITE_STEP_COMMENT_PATTERN = Pattern
+            .compile("^!--\\s+Deprecated:\\s+(.*),\\s+(.*)$");
+    private static final Object DEPRECATED_STEP_LOG_MESSAGE_KEY = CompositeStepsReplacementStoryReporter.class;
+
+    private final TestContext testContext;
+    private final ReplacementStepMonitor replacementStepMonitor;
+
+    public CompositeStepsReplacementStoryReporter(TestContext testContext,
+            ReplacementStepMonitor replacementStepMonitor)
+    {
+        this.testContext = testContext;
+        this.replacementStepMonitor = replacementStepMonitor;
```

Comment:
not sure if its good when one lifecycle reporting class depends on another lifecycle reporting class

Reply:
> not sure if its good when one lifecycle reporting class depends on another lifecycle reporting class

What's the problem in this case?
The sequence of method calls in this case is quite clear

Reply:
I would say that we add one more responsibility to monitor class that goes beyond monitoring, so its like mixing different things up

Reply:
just move `getLogMessageForRunningDeprecatedStep` to a new separate class and reuse it in both places